coralproject/talk
# Coral Reactions and Coral Plugin Love

# Coral Reactions
This PR adds a better API for creating Reactions.
Adds `coral-plugin-love`

## Building Reactions
This PR introduces `withReactions` HOC that abstracts all the logic behind handling reactions.

A minimal reaction plugin should look like this:
```js
import React from 'react';
import {withReaction} from 'coral-plugin-api';

class LoveButton extends React.Component {
  handleClick = () => {
    const {
      postReaction,
      deleteReaction,
      alreadyReacted
    } = this.props;

    if (alreadyReacted()) {
      deleteReaction();
    } else {
      postReaction();
    }
  };

  render() {
    const {count} = this.props;
    return (
      <button onClick={this.handleClick}>
        <span>Love</span>
        <span>{count > 0 && count}</span>
      </button>
    );
  }
}

export default withReaction('love')(LoveButton);
```

Yes. That's it.

## Translations, Styles and handling User behaviour
Please check the source code within `coral-plugin-love`. That's a complete example for building Reactions.

## Coral Plugin Love
Allows you to love comments ❤️ 
![out](https://cloud.githubusercontent.com/assets/1401559/25825759/8a4e129c-341a-11e7-8697-3823d21e06f3.gif)


## Test this branch
- Pull the branch
- Go to `plugins.json` and add `coral-plugin-love` both in `client` and `server`
- Run it
- Check that the love reaction is working as expected
